frame/system/src/lib.rs

```diff
@@ -923,6 +928,44 @@ impl<T: Trait> Module<T> {
 	}
 }
 
+impl<T: Trait> Module<T> {
+	pub fn spent_weight() -> Option<Weight> {
+		CurrentDispatchableWeight::get()
+	}
+
+	/// Decrease the amount of weight consumed by the current dispatchable.
+	///
+	/// This function can only meaningfully called within `Self::dispatch`. Otherwise,
+	/// it does nothing and returns `Err`.
+	///
+	/// The total weight returned from the dispatchable cannot exceed its total requested weight.
+	pub fn return_unspent_weight(unspent: Weight) -> Result<(), ()> {
+		let weight = match CurrentDispatchableWeight::take() {
```

Comment:
```suggestion
		let weight = match CurrentDispatchableWeight::take().ok_or(())?
```

(I don't see much points in doing such reviews on a _draft_ -- this is an exception)

Reply:
aha yeah agreed. I would not have been able to help myself as well if I saw something like this )